diff -Nurp a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
--- a/drivers/mmc/card/block.c	2015-08-18 16:55:18.867727345 +0530
+++ b/drivers/mmc/card/block.c	2015-08-18 17:09:15.279717944 +0530
@@ -1896,6 +1896,12 @@ static int mmc_blk_issue_rq(struct mmc_q
 	if (req && !mq->mqrq_prev->req)
 		/* claim host only for the first request */
 		mmc_claim_host(card->host);
+	if (md->area_type & MMC_BLK_DATA_AREA_RPMB) {
+		if (req)
+			blk_end_request_all(req, 0);
+		ret = 0;
+		goto out;
+	}
 
 	ret = mmc_blk_part_switch(card, md);
 	if (ret) {
