--- /home/tegradev/oe/output/work/colibri-t20-angstrom-linux-gnueabi/u-boot-git-r80/git/board/toradex/common/tegra2_nand.c~	2012-01-17 11:09:37.000000000 +0100
+++ /home/tegradev/oe/output/work/colibri-t20-angstrom-linux-gnueabi/u-boot-git-r80/git/board/toradex/common/tegra2_nand.c	2012-01-18 20:06:36.439745214 +0100
@@ -954,7 +954,8 @@
 	tmp_id = nand_read_byte(&tmp_mtd);
 	tmp_3rd = nand_read_byte(&tmp_mtd);
 	tmp_4th = nand_read_byte(&tmp_mtd);
-
+	(void)tmp_3rd;
+	
 	sprintf(compat, "%02X,%02X,%02X", tmp_manf, tmp_id, tmp_4th);
 	node = fdt_node_offset_by_compatible(gd->blob, 0, compat);
 	if (node < 0) {
